### PR TITLE
Add multi-cluster-ingress-offline document

### PR DESCRIPTION
**Pull request by luoMonkeyKing**
Signed-off-by: luoMonkeyKing luomonkeyking@gmail.com
What type of PR is this? 
/kind documentation

What this PR does / why we need it:
In the process of testing mci, due to network reasons, the mirror construction is inconvenient, and there is no relevant offline installation documentation in the community, so I have compiled a copy, hoping to help the community.

Which issue(s) this PR fixes:
Fixes #
Added mcil offline documentation

Special notes for your reviewer:
My document is in Chinese, I don't know if it is possible

Does this PR introduce a user-facing change?:
NONE

### CODE REVIEW

File: docs/multi-cluster-ingress-offline.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
+# 离线部署multi-cluster-ingress
+在已有的k8s集群上通过helm离线部署muti-cluster-ingress
+# 前期准备
```
Comment (XiShanYongYe-Chang) [inline docs/multi-cluster-ingress-offline.md:3]: 此处可以使用一个二级标题：`## 前期准备`，下面 类似的地方也是。

File: docs/multi-cluster-ingress-offline.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+//kube-webhook-certgen是官方镜像不需要自己编译
+docker pull luomonkeyking/kube-webhook-certgen:v1.1.1
+
+//hello-app的镜像是官方例子，可以根据需要下载
+docker pull luomonkeyking/hello-app:1.0
+```
+
+# 部署
+在master集群上通过helm部署，需要用到multi-cluster-ingress源码中的charts部分，所以需要将源码下载到环境中
+### 下载代码
```
Comment (XiShanYongYe-Chang) [inline docs/multi-cluster-ingress-offline.md:37]: 上面已经提供了下载代码的步骤，这里是不是可以去掉呢？

File: docs/multi-cluster-ingress-offline.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    610         ## use *either* current default registry/image or repository format or installing chart by providing the values.yaml will fai        l
+    611         ## repository:
+    612         tag: v1.1.1
+                //这里注释掉了
+    613         #digest: sha256:64d8c73dca984af206adf9d6d7e46aa550362b1d7a01f3a0a91b20cc67868660
+    614         pullPolicy: IfNotPresent
+```
+
+### 通过helm离线部署
+在multi-cluster-ingress-nginx/charts目录下进行安装
```
Comment (XiShanYongYe-Chang) [inline docs/multi-cluster-ingress-offline.md:79]: 这句话的意思是指用户需要进入到`charts`目录下么？

File: docs/multi-cluster-ingress-offline.md [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+### FAQ
+- 在测试应用的最后一步通过port-forward端口转发时没有添加--address参数时，默认走的是127.0.0.1，所以在hosts中添加域名映射时，需要加到127.0.0.1后面
+```
+127.0.0.1   localhost localhost.localdomain localhost4 demo.localdev.me
+```
+如果想要将域名映射添加到主机ip后面的话，可以在命令中添加--address参数
+
+```
+kubectl port-forward --address 0.0.0.0 --namespace=ingress-nginx service/ingress-nginx-controller 8080:80
+```
```
Comment (XiShanYongYe-Chang) [inline docs/multi-cluster-ingress-offline.md:97]: 这里是不是两种方式呢？

**Review by XiShanYongYe-Chang — COMMENTED** [review]
新增的文件名称可以考虑增加一个`-zh`的后缀。

File: docs/multi-cluster-ingress-offline.md [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+### 下载代码
+
+```
+// for HTTPS
+git clone https://github.com/karmada-io/multi-cluster-ingress-nginx.git
+// for SSH
+git clone git@github.com:karmada-io/multi-cluster-ingress-nginx.git
+```
+### 修改charts文件
+这里我把charts目录下values.yaml中的镜像digest给注释了，不然识别不了(这里不是很熟悉，欢迎了解的小伙伴补充),同时需要将上面构建的镜像通过docker tag的方式修改成和value.yaml中的一致。
```
Comment (lonelyCZ) [inline docs/multi-cluster-ingress-offline.md:46]: 这个digest应该是用来鉴别该镜像是否是官方提供的，自己构建的镜像会有不同的digest

**Comment by lonelyCZ** [comment]
我刚通过这个文档，部署成功了，非常感谢 @luoMonkeyKing

**Comment by XiShanYongYe-Chang** [comment]
> 我刚通过这个文档，部署成功了，非常感谢

Congratulations! Kindly ping @luoMonkeyKing

**Comment by luoMonkeyKing** [comment]
> 新增的文件名称可以考虑增加一个`-zh`的后缀。

好的

File: docs/multi-cluster-ingress-offline.md (hunk 1 above)
Comment (luoMonkeyKing) [reply docs/multi-cluster-ingress-offline.md:3]: 好的，我后面修改下

File: docs/multi-cluster-ingress-offline.md (hunk 2 above)
Comment (luoMonkeyKing) [reply docs/multi-cluster-ingress-offline.md:37]: 嗯，可以去掉

File: docs/multi-cluster-ingress-offline.md (hunk 3 above)
Comment (luoMonkeyKing) [reply docs/multi-cluster-ingress-offline.md:79]: 是的

File: docs/multi-cluster-ingress-offline.md (hunk 4 above)
Comment (luoMonkeyKing) [reply docs/multi-cluster-ingress-offline.md:97]: 也可以这么理解，第一种是默认方式

File: docs/multi-cluster-ingress-offline.md (hunk 5 above)
Comment (luoMonkeyKing) [reply docs/multi-cluster-ingress-offline.md:46]: 好的，多谢，这里不注释掉的话，镜像会带着digest无法下载，我把digest打进image的tag中好像也不行，就注释掉了

**Comment by Poor12** [comment]
Hi, @luoMonkeyKing , we are going to move docs to our website. Since our main repo is not suitable for Chinese documents, welcome to make a PR to https://github.com/karmada-io/website/tree/main/i18n/zh/docusaurus-plugin-content-docs/current/userguide/service.

**Comment by luoMonkeyKing** [comment]
> Hi, @luoMonkeyKing , we are going to move docs to our website. Since our main repo is not suitable for Chinese documents, welcome to make a PR to https://github.com/karmada-io/website/tree/main/i18n/zh/docusaurus-plugin-content-docs/current/userguide/service.

ok

**Comment by RainbowMango** [comment]
@XiShanYongYe-Chang Do we have the document on the website?

**Comment by XiShanYongYe-Chang** [comment]
> @XiShanYongYe-Chang Do we have the document on the website?

No. We need to move this pr on.

**Comment by RainbowMango** [comment]
@luoMonkeyKing Thanks for doing this.  But as of now, all docs are moved to karmada-io/website, as mentioned on above https://github.com/karmada-io/karmada/pull/1925#issuecomment-1207752024. 
So, I'm going to close it now.

/close

**Pull request closed without merging**